server/views/script.js

```diff
@@ -0,0 +1,224 @@
+  changeInput(cell);
+  changeCell(cell);
+}
+
+function changeInput(cell) {
+  cell.oninput = function() {
+    input.value = cell.value;
+    input.setAttribute('cellId', cell.id);
+    highlightTheMentionedCell(cell.value);
+  };
+}
+
+function highlightTheMentionedCell(value) {
+  let cells = document.getElementsByClassName('cell');
+  for (let i = 0; i < cells.length; i++) {
+    cells[i].classList.remove("highlightRed");
+  }
+  let links = value.match(/[A-Z]\d+/g);
+  if (!links) {
+    return;
+  }
+  for (let i = 0; i < links.length; i++) {
+    const letter = links[i].match(/[A-Z]/)[0];
+    const number = links[i].match(/(\d+)/)[0];
+    let id = `${number - 1}_${letter.charCodeAt(0) - 65}`;
+    let linkCell = document.getElementById(id);
+    linkCell.classList.add("highlightRed");
+  }
+}
+
+function changeCell(cell) {
+  input.oninput = function() {
+    cell.value = input.value;
+    highlightTheMentionedCell(input.value);
+  };
+}
+
+function writeCell() {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  cell.value = input.value;
+  calculate();
+  setTimeout(writeInput, 1000);
+}
+
+function writeInput(argument) {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  input.value = cell.value;
+}
+
+function backlight(id, color) {
+  let numberId = id.match(/(\d+)_/)[1];
+  let letterId = id.match(/_(\d+)/)[1];
+  let numberCell = document.getElementById(numberId + '_number');
+  let letterCell = document.getElementById(letterId + '_letter');
+  numberCell.style.backgroundColor = color;
+  letterCell.style.backgroundColor = color;
+}
+
+function addClass(element, newClass) {
+  let modal = document.getElementsByClassName(element)[0];
+  modal.classList.add(newClass);
+}
+
+function removeClass(element, newClass) {
+  let modal = document.getElementsByClassName(element)[0];
+  modal.classList.remove(newClass);
+}
+
+function download(argument) {
+  let input = document.getElementsByClassName('inputfilename')[0];
+  input.value = '';
+  addClass('filename', 'showfilename');
+  addClass('container', 'cover');
+}
+
+function showModal(argument) {
+  addClass('modal', 'showmodal');
+  addClass('container', 'cover');
+  let xhr = new XMLHttpRequest();
+  xhr.open("GET", "/lastfiles");
+  xhr.send();
+  xhr.onload = () => printFiles(xhr.response);
+}
+
+function writeFile(argument) {
+  let fileName = document.getElementsByClassName('inputfilename')[0].value;
+  calculate(fileName);
+  removeClass('filename', 'showfilename');
+  removeClass('container', 'cover');
+}
+
+function printFiles(json) {
+  let modal = document.getElementsByClassName('modal')[0];
+  html = '';
+  if (json === '{}') {
+    html += '<p class="chooseFile">No files to open</p>';  
+    modal.innerHTML = html;
+    return;
+  }
+  html += '<p class="chooseFile">Choose File</p>';
+  let records = JSON.parse(json);
+  for (let k in records) {
```

Comment:
don't use for-in
use built-in array methods

Reply:
done